tests/test_client_request.py

```diff
@@ -1635,7 +1635,17 @@ async def test_write_bytes_with_iterable_content_length_limit(
     """Test that write_bytes respects content_length limit for iterable data."""
     # Test with iterable data
     req = ClientRequest("post", URL("http://python.org/"), loop=loop)
-    req.body = data  # type: ignore[assignment]  # https://github.com/python/mypy/issues/12892
+
+    # Convert list to async generator if needed
+    if isinstance(data, list):
+
+        async def gen() -> AsyncIterator[bytes]:
+            for chunk in data:
+                yield chunk
+
+        req.body = gen()
+    else:
+        req.body = data
```

Comment:
These kind of tests are redundant, given that no actual production code will ever execute this. It'd be more useful to pass `data` in to the ClientRequest construction above, if there is still something relevant to test.

But, there's point having a test that verifies that write_bytes() works when body is a list etc. because that can never happen in actual use.

Reply:
I guess you've put a setter on it, though I wonder if we shouldn't discourage that..

Reply:
I'm working on fixing the tests, adding new tests, than I'll come back and remove the ones that don't make sense

Reply:
I think we need to keep it because we have to have backwards compatibility by allowing body to be set.

Reply:
Yeah I think if we ship this right away, its fine to drop

Reply:
Let me add a breaking note for that but qualify it as it was added in 3.12.0 so we don't expect anyone is using it after only being released for a few hours

Reply:
I'm going to make all the `update_*` methods protected as well since someone may start using them otherwise

Reply:
Yeah, I figured a subclass might overload the methods or something, so wasn't so sure on changing the methods. Making body read-only is probably fine though.

Reply:
> https://github.com/Slzdude/aiohttp-proxy/blob/8810ab6c0daf9f537f2c2bb1ce049fbdae7b9f6e/aiohttp_proxy/connector.py#L97

Oh, so connectors receive the request too, fair enough.